### PR TITLE
New: Import films from TMDb by company & Pagination work on other TMDb lists

### LINKED ISSUE #5490: Pagination not implemented for TMDB User
**Describe the bug**
 Created a TMDB watchlist and Radarr only grabs the first 20 movies. List has 60+ movies. Very similar in description and effect to  #4441

**To Reproduce**
Add TMDB watchlist to Radarr. Wait. See only the first 20 grabbed.

**Expected behavior**
Radarr should fetch all results from list (or at least many more). Doesn't look like Radarr is performing the pagination needed.

**Platform Information (please complete the following information):**
 - OS: Ubuntu 20.04
 - Docker: Yes
 - Package Version: 3.0.0.4204-ls86 by linuxserver.io 
 - Mono or.NET Core Version: Yes (3.1.10)
 - Radarr Version 3.0.0.4204

**Trace Logs**
(Uploaded full trace to https://pastebin.com/ubs5nGvr)
2020-12-4 23:07:49.7|Info|ImportListSyncService|Starting Import List Refresh for List coolmule0
2020-12-4 23:08:43.5|Debug|TMDbUserImport|Downloading List https://api.themoviedb.org/4/account/5fcaa639383df2003e594537/movie/watchlist
2020-12-4 23:08:43.5|Debug|Api|[GET] /api/v3/health: 200.OK (0 ms)
2020-12-4 23:08:45.5|Debug|FetchAndParseImportListService|Found 20 from TMDb User
2020-12-4 23:08:45.5|Debug|FetchAndParseImportListService|Found 20 reports for list TMDb User

### LINKED ISSUE #6023: Feature request: TMDb Company list option
**Is your feature request related to a problem? Please describe.**
For those people who really enjoy movies made by particular production companies, a company list option would be very useful as a counterpart to using the TMDb Person option to always get movies with particular directors or crew.

**Describe the solution you'd like**
In Settings -> List, an additional TMDb option of "TMDb Company" that works like TMDb Person, but with a CompanyID corresponding to the IDs of company pages on TMDb (such as "3" for [the Pixar company page](https://www.themoviedb.org/company/3/movie)). This would probably have no options.

**Describe alternatives you've considered**
User-maintained TMDb lists can serve a similar purposes, but their decentralized nature makes them less useful, as the vast majority of lists are unmaintained after being originally created.

### PR DESCRIPTION
**Pull request by RobinDadswell**
#### Database Migration
NO

#### Description
Allows for import of films from a TMDB company

#### Todos
- [x] Tests
- [x] Translation Keys
- [ ] Wiki Updates
- [x] Work out pagination issue (currently only does first 20 films which is the first page)

#### Issues Fixed or Closed by this PR

* Fixes #6023 
* Fixes #5490

### CODE REVIEW

File: src/NzbDrone.Core/ImportLists/TMDb/Company/TMDbCompanyRequestGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                                               .SetSegment("id", $"movie?with_companies={Settings.CompanyId}")
+                                               .SetSegment("secondaryRoute", "");
+
+            var jsonResponse = JsonConvert.DeserializeObject<MovieSearchResource>(HttpClient.Execute(requestBuilder.Build()).Content);
+
+            MaxPages = jsonResponse.TotalPages;
+
+            for (var pageNumber = 1; pageNumber <= MaxPages; pageNumber++)
+            {
+                requestBuilder = RequestBuilder.Create()
```
Comment (RobinDadswell) [inline src/NzbDrone.Core/ImportLists/TMDb/Company/TMDbCompanyRequestGenerator.cs:45]: FYI requestBuilder.AddQueryParam adds it with a ? and it needs to be an & hence why it is like this

Comment (Qstick) [reply src/NzbDrone.Core/ImportLists/TMDb/Company/TMDbCompanyRequestGenerator.cs:45]: You should be able to use add query parameters, you just need to use it for both with_companies and page. 

The company query parameter can be set before the loop, then set the page query parameter in the loop with the replace flag to true to ensure it gets overwritten on each request

**Comment by RobinDadswell** [comment]
Need to add the pagination code to lists as well - will also have a look at the other TMDb code whilst at it -

**Comment by RobinDadswell** [comment]
@Qstick  logging in this for each URL - would you prefer if it were debug/trace instead of info?

**Comment by Qstick** [comment]
Yea debug is likely a good idea here.

**Review by Qstick — APPROVED** [review]
Looks good, tested with company.

**Pull request merged**